src/viadot/orchestration/prefect/flows/azure_sql_to_adls.py

```diff
@@ -0,0 +1,85 @@
+"""Flows for downloading data from Aselite and uploading it to Azure ADLS."""
+
+from typing import Any, Dict, List, Literal
+
+from prefect import flow
+from prefect.task_runners import ConcurrentTaskRunner
+from viadot.orchestration.prefect.tasks import df_to_adls, azure_sql_to_df
+
+
+@flow(
+    name="Aselite extraction to ADLS",
+    description="Extract data from Aselite Cloud"
+    + " and load it into Azure Data Lake Storage.",
+    retries=1,
+    retry_delay_seconds=60,
+    task_runner=ConcurrentTaskRunner,
+    log_prints=True,
+)
+def azure_sql_to_adls(
+    query: str = None,
+    credentials_secret: str = None,
+    sep: str = "\t",
+    file_path: str = None,
+    if_exists: Literal["replace", "append", "delete"] = "replace",
+    validate_df_dict: Dict[str, Any] = None,
+    convert_bytes: bool = False,
+    remove_special_characters: bool = None,
+    columns_to_clean: List[str] = None,
+    adls_config_key: str | None = None,
+    adls_azure_key_vault_secret: str | None = None,
+    adls_path: str | None = None,
+    adls_path_overwrite: bool = False,
+) -> None:
+    r"""
+    Flow for downloading data from ASElite to csv file, then uploading it to ADLS.
+
+    Args:
+        query (str): Query to perform on a database. Defaults to None.
+        credentials_secret (str, optional): The name of the Azure Key Vault
+            secret containing a dictionary with ASElite SQL Database credentials.
+            Defaults to None.
+        sep (str, optional): The delimiter for the output CSV file. Defaults to "\t".
+        file_path (str, optional): Local destination path. Defaults to None.
+        if_exists (Literal, optional): What to do if the table exists.
+            Defaults to "replace".
+        validate_df_dict (Dict[str], optional): A dictionary with optional list of
+            tests to verify the output dataframe. If defined, triggers the `validate_df`
+            task from task_utils. Defaults to None.
+        convert_bytes (bool). A boolean value to trigger method df_converts_bytes_to_int
+            Defaults to False.
```

Comment:
Why convert the bytes? Is this basically casting some columns from byte to integer? Why does Azure SQL return them as bytes? Worth adding a comment explaining this here as this seems quite random.

Reply:
As I spoke with Michał WIniarski it was some Aselite issue that in one table the dtypes instead of int was bytes which was making a lot of problems, so they apply such task to convert bytes to int if convert_bytes argument is set to true

Reply:
Ok, pls this info in the docstring, but in a more generic way (without mentioning Aselite).

Reply:
[06f73b3](https://github.com/dyvenia/viadot/pull/1043/commits/06f73b306bb3bf3cae4324d865529962e05a2a58)